### ISSUE TITLE
[bug] compile mangles skill frontmatter and overwrites uncommitted edits

### ISSUE THREAD

**Issue by vlad-ryzhkov** [zernie/agent-lint#107]
Ran `vigiles init && vigiles compile` on a Claude Code harness (24 skills, 3 subagents). Three independent problems; I traced each in source.

**1a. `allowed-tools` → `tools` key rename on the compile write-path.** The read side is correct — `adopt.ts` reads `frontmatterList(fm, "allowed-tools") ?? frontmatterList(fm, "tools")`, and `scan-core.ts:270` treats `allowed-tools` as "the CC skill tool contract". But `renderSkillFrontmatter` (`src/core/compile.ts:817-819`) is the only place a skill's tool contract is emitted and it hardcodes `tools:`:

```ts
if (spec.tools && spec.tools.length > 0) {
  fm.push(`tools: ${spec.tools.join(", ")}`);
}
```

So the round-trip diverges: adopt reads `allowed-tools`, compile writes `tools`. Claude Code reads `allowed-tools` for skills — the emitted `tools:` is not the skill key, so the tool restriction is effectively lost.

**1b. CSV instead of a YAML list.** The same line emits `tools: Read, Grep, Bash` — a bare comma scalar, not a YAML sequence. As literal YAML it parses as one string, not three items. It's a hand-built string template (no `yaml.dump`), so this is a deliberate authoring choice, not a serializer default. Same pattern for subagents (`compile.ts:1036`) and `disallowedTools` (`compile.ts:1039`).

**1c. `context: fork` silently lost on adopt.** `adopt.ts:334` lists `"context"` in `SKILL_KNOWN_KEYS` (so it's excluded from the unmapped-key warning), but `adoptSkill` (`adopt.ts:376-417`) never reads `context` into the generated spec — unlike `argument-hint` / `disable-model-invocation`, which it does read. `SkillSpec.context` is fully supported and `compileSkill` renders it (`compile.ts:811`), so a hand-authored spec survives, but an existing `SKILL.md` with `context: fork` loses it on `vigiles init` — no warning, because the key is falsely marked "known". (Note: I originally suspected an `agent:` key was dropped too — that's **not** a vigiles/Claude Code skill key, so nothing is dropped there; a custom `agent:` field would simply land in the unmapped-key warning.)

**1d. Dirty-tree overwrite.** `init` generated `.spec.ts` from the on-disk state, then `compile` regenerated the `.md` over uncommitted edits, no warning, no recovery.

**Repro:** a CC skill whose frontmatter has `allowed-tools` (YAML list) + `context: fork`; `vigiles init --target=<skill>`, `vigiles compile`, diff.

**Suggested fixes:**

- `compile.ts` `renderSkillFrontmatter` (~L817): emit the skill key as a real list —
  ```diff
  -    fm.push(`tools: ${spec.tools.join(", ")}`);
  +    fm.push(`allowed-tools:`);
  +    for (const t of spec.tools) fm.push(`  - ${t}`);
  ```
  (or flow-style `allowed-tools: [${spec.tools.join(", ")}]` to stay one line — still a real sequence).
- `adopt.ts` `adoptSkill` (~L380): actually read `context` (`frontmatterScalar(fm, "context")`) and carry it into the spec, mirroring `argument-hint`.
- `init`/`compile`: refuse to overwrite files with uncommitted changes unless `--force` (or require a clean tree).
- Add a round-trip regression test (adopt → compile → same `allowed-tools`/`context` keys back); `adopt.test.ts` currently has no `context` assertion.

**Harness-safety of the fix (I checked the adapter layer so this doesn't regress non-Claude-Code targets).** The skill-key fix only touches the `profile === "claude-code"` branch of `renderSkillFrontmatter` (`compile.ts:805`). Codex / OpenCode use the `"minimal"` profile (`src/adapters/codex/dialect.ts`, `src/adapters/opencode/dialect.ts`), which emits only `name`/`description` — they never see a `tools:`/`allowed-tools:` line either way, so changing the CC branch can't break them. Two Claude-Code-specific caveats to keep the fix correct:

- Change **only the skill emitter** to `allowed-tools`. `renderAgentFrontmatter` (subagents, `compile.ts:1035`) should keep `tools:` — that _is_ the correct key for Claude Code subagents; skills and subagents legitimately differ here.
- `context: fork` (1c) is Claude-Code-only and already gated to the CC profile on emit (`compile.ts:811`), so carrying it through `adopt` is a no-op for other harnesses.

Thanks for building vigiles — happy to provide more logs or a minimal repro if useful.

**Comment by zernie** [comment]
Thanks for the source-traced report — this was exactly right on all four points. Shipped in v14.12.0:

- **1a/1b** — `renderSkillFrontmatter` now emits the skill's tool contract as `allowed-tools` (the key Claude Code actually reads for skills) as a real YAML sequence, not a `tools:` CSV scalar. Subagents correctly keep `tools:` — they legitimately differ, as you noted.
- **1c** — `adoptSkill` now reads `context` and carries it into the generated spec, so `context: fork` survives `init`.
- A round-trip regression test (adopt → compile → same `allowed-tools`/`context` back) locks it in.

**1d (dirty-tree overwrite guard) is deferred** — we're still deciding the right UX for the refuse-on-uncommitted-changes flow, so I don't want to half-ship it. Tracking it separately. Appreciate the detailed repro.

---
_Generated by [Claude Code](https://claude.ai/code)_
